### PR TITLE
Add Asio C++ library 1.21.0

### PR DESCRIPTION
**Pull request by ltekieli**

### CODE REVIEW

File: modules/asio/1.21.0/patches/add_build_file.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,9 @@
++cc_library(
```
Comment (meteorcloudy) [inline modules/asio/1.21.0/patches/add_build_file.patch:4]: Ideally this BUILD file should be upstreamed so we don't need it for future version, can you help with that?

File: modules/asio/1.21.0/presubmit.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - centos7
+  - debian10
+  - ubuntu2004
+  - macos
+  - windows
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    build_targets: []
```
Comment (meteorcloudy) [inline modules/asio/1.21.0/presubmit.yml:12]: Can you specify a build target here to make sure this module actually works?
probably:
```
build_targets:
  -- @asio//:asio
```

**Review by meteorcloudy — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: modules/asio/1.21.0/patches/add_build_file.patch (hunk 1 above)
Comment (ltekieli) [reply modules/asio/1.21.0/patches/add_build_file.patch:4]: I can try. However, I'm not sure if the author is accepting other build systems. Let's see.

Comment (meteorcloudy) [reply modules/asio/1.21.0/patches/add_build_file.patch:4]: Thanks!

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**